emukit/core/optimization/acquisition_optimizer.py

```diff
@@ -69,10 +69,15 @@ def f_df(x):
 
         if acquisition.has_gradients:
             _log.info("Starting gradient-based optimization of acquisition function {}".format(type(acquisition)))
-            return self.gpyopt_acquisition_optimizer.optimize(f, None, f_df)
+            x, f_min = self.gpyopt_acquisition_optimizer.optimize(f, None, f_df)
         else:
             _log.info("Starting gradient-free optimization of acquisition function {}".format(type(acquisition)))
-            return self.gpyopt_acquisition_optimizer.optimize(f, None, None)
+            x, f_min = self.gpyopt_acquisition_optimizer.optimize(f, None, None)
+
+        rounded_x = self.space.round(x)
+        rounded_f_min = acquisition.evaluate(rounded_x)
+
+        return rounded_x, rounded_f_min
 
     def _validate_context_parameters(self, context):
```

Comment:
can you add a small comment about what why this is needed?

Reply:
Sure. Although I've butchered diff of this PR with merging develop, so you'll have to take my word for it.

Reply:
In fact I take it back, you can take a look at the diff for the last commit, which just adds the comment